tests/orm/node/test_node.py

```diff
@@ -908,3 +908,20 @@ def test_open_wrapper():
     iter(node.open(filename))
     node.open(filename).__next__()
     node.open(filename).__iter__()
+
+
+@pytest.mark.usefixtures('clear_database_before_test')
+def test_uuid_equality_fallback():
+    """Tests the fallback mechanism of checking equality by comparing uuids."""
+    node_0 = Data().store()
+
+    nodepk = Data().store().pk
+    node_a = load_node(pk=nodepk)
+    node_b = load_node(pk=nodepk)
+
+    assert node_a == node_b
+    assert node_a != node_0
+    assert node_b != node_0
+
+    # Also test the hashing
+    assert {node_0, node_a, node_b} == {node_0, node_a}
```

Comment:
I think it would be better to explicitly test the hashing with `hash(node_0)` etc.

In the sets, all nodes could in principle have the same hash, in which case `__eq__` is used to ultimately determine if they are the same.